query-engine/connectors/sql-query-connector/src/query_builder/read.rs

```diff
@@ -67,10 +67,57 @@ where
     columns.fold(query.into_select(model), |acc, col| acc.column(col))
 }
 
-pub fn count_by_model(model: &ModelRef, query_arguments: QueryArguments) -> Select<'static> {
-    let selected_columns = model.primary_identifier().as_columns();
-    let base_query = get_records(model, selected_columns, query_arguments);
-    let table = Table::from(base_query).alias("sub");
+pub fn aggregate(model: &ModelRef, aggregators: &[Aggregator], args: QueryArguments) -> Select<'static> {
```

Comment:
It's not really clear what the aliasing does here and what kind of query you're generating. I'm thinking of could the functions used in `aggregate` be in the models or quaint, but it's not that easy to understand the resulting query over here.

Maybe write a rustdoc for the function explaining what it does and we can process it further.